datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/PatchEntitiesResolver.java

```diff
@@ -0,0 +1,209 @@
+      final PatchEntityInput input = inputs.get(i);
+      final Urn entityUrn = entityUrns.get(i);
+
+      try {
+        // Validate aspect exists
+        final var aspectSpec =
+            _entityRegistry
+                .getEntitySpec(entityUrn.getEntityType())
+                .getAspectSpec(input.getAspectName());
+        if (aspectSpec == null) {
+          throw new IllegalArgumentException(
+              "Aspect "
+                  + input.getAspectName()
+                  + " not found for entity type "
+                  + entityUrn.getEntityType());
+        }
+
+        // Always use patch aspect - same as single patchEntity resolver
+        GenericAspect aspect =
+            PatchResolverUtils.createPatchAspect(
+                input.getPatch(),
+                input.getArrayPrimaryKeys(),
+                input.getForceGenericPatch(),
+                context);
+
+        // Create MetadataChangeProposal
+        final com.linkedin.mxe.MetadataChangeProposal mcp =
+            PatchResolverUtils.createMetadataChangeProposal(
+                entityUrn,
+                input.getAspectName(),
+                aspect,
+                input.getSystemMetadata(),
+                input.getHeaders());
+
+        mcps.add(mcp);
+
+        log.debug("Created MCP for input {}: {}", i, entityUrn);
+      } catch (Exception e) {
+        log.error("Failed to create MCP for input {} ({}): {}", i, entityUrn, e.getMessage(), e);
+        throw new RuntimeException(
+            "Failed to create MCP for input " + i + ": " + e.getMessage(), e);
+      }
+    }
+
+    // Apply all patches using GraphQL standard batch approach
+    try {
+      Urn actor = UrnUtils.getUrn(authentication.getActor().toUrnStr());
+      EntityUtils.ingestChangeProposals(
+          context.getOperationContext(),
+          mcps,
+          _entityService,
+          actor,
+          false); // synchronous for GraphQL
+      log.debug("Successfully applied batch patch for {} entities", mcps.size());
+    } catch (Exception e) {
+      log.error("Failed to apply batch patch for {} entities: {}", mcps.size(), e.getMessage(), e);
+      throw new RuntimeException("Failed to apply batch patch: " + e.getMessage(), e);
+    }
+
+    // Map results back to input order
+    final List<PatchEntityResult> patchResults = new ArrayList<>();
+    for (int i = 0; i < inputs.size(); i++) {
+      final PatchEntityInput input = inputs.get(i);
+      final Urn resolvedUrn = entityUrns.get(i);
+
+      // Extract entity name from the patch operations
+      String entityName = PatchResolverUtils.extractEntityName(input.getPatch());
+
+      // Validate name for glossary entities
+      PatchResolverUtils.validateNameForEntityType(input.getEntityType(), entityName);
+
+      // Since we used batch processing, all entities succeeded if we reached this point
+      // Use resolved URN instead of input URN (which may be null for auto-generated)
```

Comment:
What do you mean the urn is auto-generated or null? You can't apply a patch to an aspect without the urn?